src/browser/api/system.js

```diff
@@ -248,7 +248,23 @@ exports.System = {
         return hash.digest('hex');
     },
     getDeviceId: function() {
-        return electronApp.getHostToken();
+        if (process.platform === 'win32') {
+            return electronApp.getHostToken();
+        } else {
+            try {
```

Comment:
No need to `try...catch` . The exception gets handled correctly (Nacked) before it goes back to the javascript-adapter.

Reply:
Removed try/catch